exchanges/bybit/bybit.go

```diff
@@ -8,740 +8,2268 @@ import (
+	params.Set("category", category)
+	if side == "" {
+		return nil, order.ErrSideIsInvalid
+	}
+	params.Set("side", side)
+	var resp *BorrowQuota
+	return resp, by.SendAuthHTTPRequestV5(ctx, exchange.RestSpot, http.MethodGet, "/v5/order/spot-borrow-check", params, nil, &resp, defaultEPL)
+}
 
-	return resp.Data.IsCancelled, by.SendAuthHTTPRequest(ctx, exchange.RestSpot, http.MethodDelete, bybitFastCancelSpotOrder, params, nil, &resp, privateSpotRate)
+// SetDisconnectCancelAll You can use this endpoint to get your current DCP configuration.
+// Your private websocket connection must subscribe "dcp" topic in order to trigger DCP successfully
+func (by *Bybit) SetDisconnectCancelAll(ctx context.Context, arg *SetDCPParams) error {
+	if arg == nil {
+		return errNilArgument
+	}
+	if arg.TimeWindow == 0 {
+		return errDisconnectTimeWindowNotSet
+	}
+	return by.SendAuthHTTPRequestV5(ctx, exchange.RestSpot, http.MethodPost, "/v5/order/disconnected-cancel-all", nil, arg, &struct{}{}, defaultEPL)
 }
 
-// BatchCancelOrder cancels orders in batch based upon symbol, side or orderType
-func (by *Bybit) BatchCancelOrder(ctx context.Context, symbol, side, orderTypes string) (bool, error) {
+// GetPositionInfo retrieves real-time position data, such as position size, cumulative realizedPNL.
+func (by *Bybit) GetPositionInfo(ctx context.Context, category, symbol, baseCoin, settleCoin, cursor string, limit int64) (*PositionInfoList, error) {
+	if category == "" {
+		return nil, errCategoryNotSet
+	} else if category != cLinear && category != cInverse && category != cOption {
+		return nil, fmt.Errorf("%w, category: %s", errInvalidCategory, category)
+	}
+	if symbol == "" && settleCoin == "" {
+		return nil, errSymbolOrSettleCoinRequired
+	}
 	params := url.Values{}
 	if symbol != "" {
 		params.Set("symbol", symbol)
 	}
-	if side != "" {
-		params.Set("side", side)
+	params.Set("category", category)
+	if baseCoin != "" {
+		params.Set("baseCoin", baseCoin)
+	}
+	if settleCoin != "" {
+		params.Set("settleCoin", settleCoin)
+	}
+	if cursor != "" {
+		params.Set("cursor", cursor)
+	}
+	if limit > 0 {
+		params.Set("limit", strconv.FormatInt(limit, 10))
+	}
+	var resp *PositionInfoList
+	return resp, by.SendAuthHTTPRequestV5(ctx, exchange.RestSpot, http.MethodGet, "/v5/position/list", params, nil, &resp, getPositionListEPL)
+}
+
+// SetLeverageLevel sets a leverage from 0 to max leverage of corresponding risk limit
+func (by *Bybit) SetLeverageLevel(ctx context.Context, arg *SetLeverageParams) error {
+	if arg == nil {
+		return errNilArgument
+	}
+	switch arg.Category {
+	case "":
+		return errCategoryNotSet
+	case cLinear, cInverse:
+	default:
+		return fmt.Errorf("%w, category: %s", errInvalidCategory, arg.Category)
+	}
+	if arg.Symbol == "" {
+		return errSymbolMissing
+	}
+	switch {
+	case arg.BuyLeverage <= 0:
+		return fmt.Errorf("%w code: 10001 msg: invalid buy leverage %f", errInvalidLeverage, arg.BuyLeverage)
+	case arg.BuyLeverage != arg.SellLeverage:
```

Comment:
Funny how this actually works but the docs says it shouldn't. I won't get you to change this rn
```
Bybit request body: {"category":"linear","symbol":"BTCUSDT","buyLeverage":"4","sellLeverage":"10"}
--- PASS: TestSetLeverageLevel (2.00s)
HTTP status: 200 OK, Code: 200
Bybit raw response: {"retCode":0,"retMsg":"OK","result":{},"retExtInfo":{},"time":1700445584049}
```